keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/support/ExtensionRegistry.kt

```diff
@@ -0,0 +1,26 @@
+package com.netflix.spinnaker.keel.api.support
+
+/**
+ * Registers an extension to a model so that it may be used in a delivery config.
+ */
+interface ExtensionRegistry {
+  fun <BASE : Any> register(
+    baseType: Class<BASE>,
+    extensionType: Class<out BASE>,
+    discriminator: String
+  )
+
+  fun <BASE : Any> extensionsOf(baseType: Class<BASE>): Map<String, Class<out BASE>>
+
+  fun baseTypes(): Collection<Class<*>>
+}
+
+inline fun <reified BASE : Any> ExtensionRegistry.extensionsOf(): Map<String, Class<out BASE>> =
```

Comment:
Just curious: is there any advantage to making these extension functions instead of `default` functions inside the `interface` body?

Reply:
You can't inline a member function, and if it's not inline it can't use reified generics.

Reply:
Ah. Never mind, then. 🙂